ncm-systemd/src/main/perl/Systemd/Service.pm

```diff
@@ -0,0 +1,519 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+# ${build-info}
+
+package NCM::Component::Systemd::Service;
+
+use 5.10.1;
+use strict;
+use warnings;
+
+use parent qw(CAF::Object Exporter);
+
+use NCM::Component::Systemd::Service::Chkconfig;
+use NCM::Component::Systemd::Service::Unit qw(:states :types);
+use NCM::Component::Systemd::Systemctl qw(systemctl_command_units);
+
+use LC::Exception qw (SUCCESS);
+
+use Readonly;
+
+# these won't be turned off with default settings
+# TODO add some systemd services? systemd itself? rc.local?
+# TODO protecting network assumes ncm-network is being used
+# TODO shouldn't these services be "always on"?
+Readonly::Hash my %DEFAULT_PROTECTED_SERVICES => (
+    network => 1,
+    messagebus => 1,
+    haldaemon => 1,
+    sshd => 1,
+);
+
+Readonly my $BASE => "/software/components/systemd";
```

Comment:
hmm, it makes things complicated on my end as `Systemd::Service` is not a component. i'll have to think about this.

the current usage i have in mind is to use unmodified `ncm-chkconfig` profiles under EL7 by including the following template in EL7:
https://github.com/stdweird/configuration-modules-core/blob/systemd/ncm-systemd/src/main/pan/components/systemd/legacy/chkconfig.pan

it sets an alias on `ncm-chkconfig` for a partial `ncm-systemd`, namely the part of `ncm-systemd` that deals with services. (`ncm-systemd` will also be able to do e.g. unit configuration, the equivalent of current `ncm-sysconfig`).